helpers/network-recorder.js

```diff
@@ -21,14 +21,20 @@ const NetworkManager = require('./web-inspector').NetworkManager;
 const REQUEST_FINISHED = NetworkManager.EventTypes.RequestFinished;
 
 class NetworkRecorder {
-  constructor(recordArray) {
+  constructor(recordArray, requestListener) {
     this._records = recordArray;
 
+    this._listener = requestListener;
+
     this.networkManager = NetworkManager.createWithFakeTarget();
 
     // TODO(bckenny): loadingFailed calls are not recorded in REQUEST_FINISHED.
     this.networkManager.addEventListener(REQUEST_FINISHED, request => {
       this._records.push(request);
+
+      if (this._listener) {
```

Comment:
I think you could have NetworkRecorder extend EventTarget/EventEmitter or just reuse the emitter inside of webinspector, so that we use real events for this instead of callbacks. (I'm expecting we'll have multiple listeners for the end of requests soon)